### ISSUE TITLE
Top navigation group title appears as clickable in mobile view

### ISSUE THREAD

**Issue by GCHQ-Developer-847** [mi6/ic-ui-kit#3259]
## Summary of the bug 
When you open a top navigation component which has grouped navigation in mobile view, the title of groups appears to be clickable when it actually isn't.

## 🪜 How to reproduce 
1. Go to the top navigation "With grouped navigation" story.
2. Make the window smaller so the top nav goes to the collapsed state and click on the button to open it.
3. Hover over and click on the title of one of the navigation groups. 
4. See that the title has a hover and pressed state (although it cannot be tabbed to).

**Note:** When using a screen reader (I used VoiceOver), you cannot tab to the title, but you can use the virtual cursor to get to it, and it is read out as a button even though it doesn't do anything. It is also says that it is "expanded".

## 📸 Screenshots or code 
https://github.com/user-attachments/assets/b7d9fc9b-1e09-4c3a-90de-66db571ad92f

## 🧐 Expected behaviour 
As the `expandable` prop has not been set on the navigation group, the title should not be clickable at all. When using a screen reader's virtual cursor, it should just read out the title as normal text, and not say that it is "expanded"; otherwise the screen reader user may think that it is clickable.
